doc/source/overview/dev_practices.rst

```diff
@@ -194,4 +194,372 @@ should not wait until a minor release. These are the steps for a patch release:
+during the almost trivial compile time. Coverage is defined as parts of the
+executable and usable source that are tested by unit tests, and the coverage for
+your library can be viewed using the ``pytest-cov`` library. For example::
+
+  $ pytest --cov numpydoc_validation
+   ============================= test session starts ==============================
+   platform linux -- Python 3.8.10, pytest-6.2.5, py-1.11.0, pluggy-1.0.0
+   rootdir: /home/user/python/numpydoc_validation
+   plugins: cov-3.0.0
+   collected 1 item
+
+   tests/test_validate.py .                                                 [100%]
+
+   ---------- coverage: platform linux, python 3.8.10-final-0 -----------
+   Name                               Stmts   Miss  Cover
+   ------------------------------------------------------
+   numpydoc_validation/__init__.py        2      0   100%
+   numpydoc_validation/_validate.py      69      0   100%
+   ------------------------------------------------------
+   TOTAL                                 71      0   100%
+
+While 100% coverage is ideal, the law of diminishing returns often applies to
+the coverage of a Python library, and it is often sufficient, to achieve 80-90%
+coverage. For parts of your library that are difficult or impossible to test,
+consider using ``# pragma: no cover`` at the end of method definition, branch,
+or line to denote that part of the code cannot be reasonably tested.  For
+example, if part of your module performs a simple ``import`` test of
+``matplotlib`` and raises an error when the library is not installed, it is not
+reasonable to attempt to test this and assume full coverage:
+
+.. code:: python
+
+   try:
+       import matplotlib
+   except ImportError:  # pragma: no cover
+       raise ImportError("Please install matplotlib to use this feature")
+
+.. note::
+   You should only avoid coverage of parts of your library where you cannot
+   reasonably test without an extensive testing suite or setup.  Most methods
+   and classes, including edge cases, can be reasonable tested. Even parts of
+   your code that raise errors like ``TypeError`` or ``ValueError`` when users
+   input the wrong data type or value.
+
+
+Unit Testing within CI/CD
+~~~~~~~~~~~~~~~~~~~~~~~~~
+Effective CI/CD assumes that unit testing is developed during feature
+development or bug fixes. However, given the limited scope of the local
+development environment, it is often not possible to enforce testing on multiple
+platforms, or even, unit testing in general. However, with the right automated
+CI/CD, such testing can still occur and be enforced automatically.
+
+`GitHub Actions <gh actions link>`_ is the preferred automated CI/CD platform
+for running Python library unit tests for PyAnsys, and can be employed
+immediately by closing the project `template <link to
+github.com/pyansys/template>`_. For a basic overview of those unfamiliar of
+GitHub Actions, read: `missing link <missing_link>`_.
+
+**Sample Workflow**
```

Comment:
If we are referring to the following sections, this should be a heading with the following entries as subheadings.